src/core/constants/async-input.d.ts

```diff
@@ -0,0 +1,4 @@
+export interface AsyncInput {
```

Comment:
What is this for? Is this a redeclaration of the exact type defined in src/core/constants/async-input?

Reply:
The type defined in the JS is for CC, includes extra function implementations to make the type-checker happy, etc. We only need this bare-bones type definitions file since it's only an interface. The code in the JS version has been removed.